server/src/test/java/org/elasticsearch/common/util/concurrent/AsyncIOProcessorTests.java

```diff
@@ -39,7 +39,7 @@ public void testPut() throws InterruptedException {
             protected void write(List<Tuple<Object, Consumer<Exception>>> candidates) throws IOException {
                 if (blockInternal) {
                     synchronized (this) {
-                        for (Tuple<Object, Consumer<Exception>> c :candidates) {
+                        for (int i = 0; i < candidates.size(); i++) {
```

Comment:
Not sure, addAndGet is used in the "else" branch of the statement in L41, maybe its there for a purpose. I didn't go into the logic of the test too much to be honest, but I thought it better to keep it as is.

Reply:
Well, it adds 1 for `candidates.size()` times which is functionally identical to add it directly without the loop.

Reply:
I agree and I added a TODO to re-check this, but I would prefer not to change it with this PR since I don't fully understand the test. Might be we can remove it though.

Reply:
Sounds good to me. Thanks for adding this.